gowiki/cache/cache.go

```diff
@@ -0,0 +1,111 @@
+package cache
+
+import (
+	"crypto/sha256"
+	"errors"
+	"time"
+
+	"discord_bot/gowiki/models"
+)
+
+var (
+	CacheExpiration time.Duration = 12 * time.Hour // Max time that a request can exist in the cache
+	MaxCacheMemory  int           = 500            // Max request can exist in the cache
+)
+
+// Find and delete string s in string slice
+func FindAndDel(arr []string, s string) []string {
+	index := 0
+	for i, v := range arr {
+		if v == s {
+			index = i
+			break
+		}
+	}
+	return append(arr[:index], arr[index+1:]...)
+}
+
+/*
+Create a cache that store:
+
+- Key: API request URL
+
+- Value: RequestResponse
+*/
+func MakeWikiCache() WikiCache {
+	res := WikiCache{}
+	res.Memory = map[string]models.RequestResult{}
+	res.HashedKeyQueue = make([]string, 0, MaxCacheMemory)
+	res.CreatedTime = map[string]time.Time{}
+	return res
+}
+
+// Cache to store Wikipedia request result
+type WikiCache struct {
+	Memory         map[string]models.RequestResult // Map store request result
+	HashedKeyQueue []string                        // Key queue. Delete the first item if reach max cache
+	CreatedTime    map[string]time.Time            // Map store created time
+}
```

Comment:
```suggestion
// WikiCache stores Wikipedia request result.
type WikiCache struct {
    // Memory map stores request result.
	Memory         map[string]models.RequestResult
	// HashedKeyQueue. Delete the first item if reach max cache.
	HashedKeyQueue []string
	 // CreatedTime map stores created time.
	CreatedTime    map[string]time.Time
}
```
Nice you included doc comments. One thing to note is that, by convention, the comments always start with name of what they are commenting and end with a dot.